components/api/src/main/java/com/hotels/styx/api/HttpRequest.java

```diff
@@ -63,7 +63,41 @@
 import static java.util.stream.Stream.concat;
 
 /**
- * HTTP request with a fully aggregated/decoded body.
+ * A HTTP request object with a byte stream body.
+ * <p>
+ * A {@code HttpRequest} is used in {@link HttpInterceptor} where each content
+ * chunk must be processed as they arrive. It is also useful for dealing with
```

Comment:
Singular/plural mismatch. Should be "each content chunk must be processed as **it** arrive**s**".